kong/tracing/propagation.lua

```diff
@@ -589,9 +589,15 @@ local function set(conf_header_type, found_header_type, proxy_span, conf_default
 
   found_header_type = found_header_type or conf_default_header_type or "b3"
 
+  -- contains all the different formats of the current request's trace ID
+  local trace_id_formats = {}
```

Comment:
> what the table contains is one or more trace_ids (possibly in different formats)

not exactly. At this point (in `propagation.set()`) the trace id has already been decided and is only one (proxy_span.trace_id), it was chosen from the incoming tracing header who "won", or just generated if there was no incoming tracing header in the request. This table contains all the formats that the current trace id can be outputed as, which depend on what tracing headers are being injected in the outgoing request. I will improve the comment to make it more clear.

You are right about b3, the "format" of the trace ID would be the same so it would be redundant to include both.